### PR TITLE
Add React Native support

### PR DESCRIPTION
**Pull request by 39otrebla**

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
I don't know if it makes sense for this library to globally shim buffer

**Comment by dcousens** [comment]
Relevant https://github.com/bitcoinjs/bitcoinjs-lib/issues/582#issuecomment-280561453

File: make-hash.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 'use strict'
+if (typeof Buffer === 'undefined') {
+  global.Buffer = require('buffer').Buffer
```
Comment (dcousens) [inline make-hash.js]: NACK,  this is not the place for this

Comment (39otrebla) [inline make-hash.js]: I have already patched bitcoinjs-lib [here](https://github.com/39otrebla/react-native-bitcoinjs-lib), but RN packager was still complaining about "Cannot find variable Buffer". I am not sure why all modules recognize Buffer but yours, but it is what happened. And I solved only by applying the "shim" for Buffer directly there, as above.

Comment (dcousens) [inline make-hash.js]: @39otrebla why not just globally patch `Buffer` in your application?

Comment (39otrebla) [inline make-hash.js]: Indeed, I've done it by importing the shim.js file as explained [here](https://github.com/39otrebla/react-native-bitcoinjs-lib). And the global patch works fine with all modules but this. I am not sure why, the only thing I came up with is the 'use-strict' directive. But I couldn't find an actual reason why 'use-strict' should "break" the patch.

Comment (dcousens) [inline make-hash.js]: Have you reported that as a bug to `react-native`?

**Comment by dcousens** [comment]
@39otrebla did you complain to `react-native`?
Many others have had this same issue,  and we can't resolve it by patching every dependency :(

**Comment by 39otrebla** [comment]
thank you for you interest @dcousens . I found the solution. It is neither a `createHash` nor a `react-native` problem. Indeed, was really strange that shimming Buffer was raising problems only with this module (I have >10 crypto-dependencies in my project, and all of them require Buffer in some way). So I haven't opened an issue on react-native, and I closed this one. Sorry for the time spent on this.

**Comment by dcousens** [comment]
@jaume

**Pull request closed without merging**